### PR TITLE
Issue 21781 add ability to create unique field values per site

### LINKED ISSUE #21781: Add Ability to Create Unique Field Values Per Site
**Is your feature request related to a problem? Please describe.**
In some cases, when customers wish to use a global content type that contains some unique field, they would like that field to be unique only for the Host they are creating on at the time, rather than globally. Currently, if marked unique the field must be unique across all hosts. 
**Describe the solution you'd like**

Change the existing Unique property to "Unique across sites" and add a new property "Unique per site".

**Additional context**
This has been requested by a customer: https://dotcms.zendesk.com/agent/tickets/106676

### PR DESCRIPTION
**Pull request by freddyucv**
Add host filter to unique validation lucene query

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java [hunk 1]
```diff
@@ -994,6 +1000,311 @@ public void selfRelatedContents() throws DotDataException {
...
+    }
+
+    /**
+     * Method to test: {@link ContentletAPI#checkin(Contentlet, User, boolean)}  }
+     * When:
+     * - Create a {@link ContentType} with a unique field
+     * - Create two  {@link Contentlet} with the same value in the unique field in different hosts
+     * - set the unique_per_site properties to false
+     *
+     * Should: Save successfully the two {@link Contentlet}
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java:1157]: bad `Should` - it should be error expected

**Comment by freddyucv** [comment]
> @freddyucv quick question: What happens if the customer enables this config prop, and then decide to disable it at some point? That'd mean they can end up with several contents with the same unique field, which would require fixing them via SQL queries.

I testes:

- Copy the default site to create second.default site
- Create a content type with two fields: one a Unique text field, and the second one a Host/Site filed
- Created two contentlet one in default site and the another in second.default site with the same value in the Text Field
- Shutdown dotCMS and set unique_per_site to false, later start up dotCMS again.
- Run reindex and everything was ok.
- Open one on the contentlet and try to save it without any changes, it fail because of duplicated value in a unique field, then change the Text Field and it was saved right

**Comment by jcastro-dotcms** [comment]
Thanks for the clarification @freddyucv ! Beyond that, can you please improve the error message in the UI and/or add a more descriptive one in the log? This problem is very easy to fix, a better error message will definitely help customers do that instead of they having to file Support a ticket for assistance.

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java]: done https://github.com/dotCMS/core/pull/21843/commits/e20c14f55318f719c36d72a7c953bf8d9ac42dfb#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R1156

**Comment by freddyucv** [comment]
> Thanks for the clarification @freddyucv ! Beyond that, can you please improve the error message in the UI and/or add a more descriptive one in the log? This problem is very easy to fix, a better error message will definitely help customers do that instead of they having to file Support a ticket for assistance.

done 

https://github.com/dotCMS/core/pull/21843/commits/e20c14f55318f719c36d72a7c953bf8d9ac42dfb#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94R2121

https://github.com/dotCMS/core/pull/21843/commits/e20c14f55318f719c36d72a7c953bf8d9ac42dfb#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R6782

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 2]
```diff
@@ -6843,6 +6856,25 @@ else if(field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())){
...
+        return String.format("Value of Field [%s] must be unique. Contents having the same value (%s): %s",
+                field.getVelocityVarName(), fieldValue, contentletSearch.getIdentifier());
+    }
+
+    public static void setUniquePerSite(final boolean uniquePerSite){
+        ESContentletAPIImpl.uniquePerSite = uniquePerSite;
+    }
+
+    public static boolean getUniquePerSite(){
+        return uniquePerSite;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:6871]: what happen if it is null?

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 3]
```diff
@@ -244,6 +246,11 @@
 
     private static final String backupPath = ConfigUtils.getBackupPath() + File.separator + "contentlets";
 
+    private static final Lazy<Boolean> UNIQUE_PER_SITE_CONFIG = Lazy.of(()->Config
+            .getBooleanProperty("unique_per_site", true));
+
+    private static Boolean uniquePerSite;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:252]: Probably better idea to use AtomicBoolean

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:6873]: I turn it to Boolean 

https://github.com/dotCMS/core/pull/21843/commits/0396a20c99adfbe928755457c8631b48e9053ba8#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R6872

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**